### PR TITLE
Fixing null pointer exception on span count sampler

### LINKED ISSUE #12745: Span count sampler throws null pointer exception
**Describe the bug**
Span count sampler is throwing null pointer exception when used alongside with an AND policy in the tailsampling processor

**Steps to reproduce**
Configure tailsampling processor to utilize an AND policy in conjunction with Span count sampler.

**What did you expect to see?**
No errors.

**What did you see instead?**
Null pointer exception.

**What version did you use?**
v0.54

**Additional context**
I'm creating a PR to fix the issue.

### PR DESCRIPTION
**Pull request by DavidPessoa**
**Description:** 
Fixing a null pointer exception bug that happens when the span_count policy is used in the tailsamplingprocessor

Fixes #12745

**Testing:** 
No tests were added

**Documentation:** 
No changes were made to the documentation

### CODE REVIEW

**Comment by DavidPessoa** [comment]
@TylerHelmuth Could you please review this PR?
Thank you!

**Review by TylerHelmuth — COMMENTED** [review]
@DavidPessoa please add a unit test that covers the bug that is being fixed.

File: processor/tailsamplingprocessor/internal/sampling/policy.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ type TraceData struct {
 	DecisionTime time.Time
 	// SpanCount track the number of spans on the trace.
 	SpanCount *atomic.Int64
-	// ReceivedBatches stores all the batches received for the trace.
+	// ReceivedBatches stores all the traces received for the batch.
```
Comment (pmm-sumo) [inline processor/tailsamplingprocessor/internal/sampling/policy.go:37]: The change is not accurate. `ReceivedBatches` contains data for a single trace, received over numerous batches (as this was described before)

**Comment by DavidPessoa** [comment]
@pmm-sumo Thank you once more for the review. 
I've updated the PR.

**Comment by TylerHelmuth** [comment]
@DavidPessoa are you still interested in this PR?

**Comment by DavidPessoa** [comment]
Yes I am @TylerHelmuth.
I've just pushed to it.

**Comment by TylerHelmuth** [comment]
> please add a unit test that covers the bug that is being fixed.

@DavidPessoa are you able to address this?

**Comment by DavidPessoa** [comment]
@TylerHelmuth yes I can

**Comment by codeboten** [comment]
Please add "Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/12745" to the description 👍🏻

**Comment by jpkrohling** [comment]
Once this is ready to be reviewed, let me know.

**Comment by DavidPessoa** [comment]
@jpkrohling I believe I'm done could you check it?
What I did was adding the tests around the `and_helper.go` file since it was where I was getting the null pointer exception

**Comment by DavidPessoa** [comment]
@jpkrohling I'm not sure how to fix the linting error...
Can you give me some suggestion on how to do it?

**Comment by DavidPessoa** [comment]
I believe it's good to go 😃

**Review by TylerHelmuth — APPROVED** [review]

**Comment by DavidPessoa** [comment]
@jpkrohling Can you have a look?

**Comment by jpkrohling** [comment]
I just rebased this PR. Let's see if we get the green light from the CI checks.

**Comment by DavidPessoa** [comment]
Looks good I believe

**Pull request merged**